ccp-openshift-scan/scanning/container-capabilities.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# this module checks the image's RUN label and provide security concerns
+# based on flags used in `docker run` command.
+# the check args method is taken from
+# github.com/projectatomic/atomic/blob/master/Atomic/backends/_docker.py
+
+import sys
+
+
+def check_args(cmd):
+    """
+    Check the arguments of the docker run command and provide feedback
+    """
+    found_sec_arg = False
+    security_args = {
+        '--privileged':
+            'This container runs without separation and should be '
+            'considered the same as root on your system.',
+        '--cap-add':
+            'Adding capabilities to your container could allow processes '
+            'from the container to break out onto your host system.',
+        '--security-opt label:disable':
+            'Disabling label separation turns off tools like SELinux and '
+            'could allow processes from the container to break out onto '
+            'your host system.',
+        '--security-opt label=disable':
+            'Disabling label separation turns off tools like SELinux and '
+            'could allow processes from the container to break out onto '
+            'your host system.',
+        '--net=host':
+            'Processes in this container can listen to ports (and '
+            'possibly rawip traffic) on the host\'s network.',
+        '--pid=host':
+            'Processes in this container can see and interact with all '
+            'processes on the host and disables SELinux within the '
+            'container.',
+        '--ipc=host':
+            'Processes in this container can see and possibly interact '
+            'with all semaphores and shared memory segments on the host '
+            'as well as disables SELinux within the container.'
+    }
+
+    for sec_arg in security_args:
+        if sec_arg in cmd:
+            if not found_sec_arg:
+                print("\nThis container uses privileged "
+                      "security switches:")
+            print("\nINFO: {}\n\t{}".format(sec_arg, security_args[sec_arg]))
+            found_sec_arg = True
+    if found_sec_arg:
+        print("\nFor more information on these switches and their "
+              "security implications, consult the manpage for "
+              "'docker run'.\n")
+
+
+def run_scan(command):
+    """
+    Runs the container capabilities can and prints error message
+    if provided command is empty
+    """
+    if not command:
+        print ("\n RUN label is not available in image under test.")
```

Comment:
How does `if not command` check for `RUN` label? Am I missing something here?

Reply:
Check this [line](https://github.com/dharmit/ccp-openshift/blob/master/seed-job/template.yaml#L103), if no run label is present in the image, we pass an empty string. Take a look at [this](https://github.com/dharmit/ccp-openshift/blob/master/seed-job/template.yaml#L103) build and container capabilities scanner's output for an image where RUN label is not available in image.